demo/shared/src/main/scala/games/utils/SimpleOBJParser.scala

```diff
@@ -0,0 +1,810 @@
+              case None => {
+                texInfo.offset = new Vector3f(x, y, 0)
+                currentShift += 3
+              }
+              case Some(z) => {
+                texInfo.offset = new Vector3f(x, y, z)
+                currentShift += 4
+              }
+            }
+          }
+        }
+
+        case "-s" if (remaining >= 2) => {
+          val x = tokens(currentShift + 1).toFloat
+
+          (if (remaining >= 3) parseFloat(tokens(currentShift + 2)) else None) match {
+            case None => {
+              texInfo.resize = new Vector3f(x, 1, 1)
+              currentShift += 2
+            }
+            case Some(y) => (if (remaining >= 4) parseFloat(tokens(currentShift + 3)) else None) match {
+              case None => {
+                texInfo.resize = new Vector3f(x, y, 1)
+                currentShift += 3
+              }
+              case Some(z) => {
+                texInfo.resize = new Vector3f(x, y, z)
+                currentShift += 4
+              }
+            }
+          }
+        }
+
+        case "-t" if (remaining >= 2) => {
+          val x = tokens(currentShift + 1).toFloat
+
+          (if (remaining >= 3) parseFloat(tokens(currentShift + 2)) else None) match {
+            case None => {
+              texInfo.turbulence = new Vector3f(x, 0, 0)
+              currentShift += 2
+            }
+            case Some(y) => (if (remaining >= 4) parseFloat(tokens(currentShift + 3)) else None) match {
+              case None => {
+                texInfo.turbulence = new Vector3f(x, y, 0)
+                currentShift += 3
+              }
+              case Some(z) => {
+                texInfo.turbulence = new Vector3f(x, y, z)
+                currentShift += 4
+              }
+            }
+          }
+        }
+
+        case "-texres" if (remaining >= 2) => {
+          texInfo.resolution = Some(tokens(currentShift + 1).toInt)
+          currentShift += 2
+        }
+
+        case _ => {
+          texInfo.path = tokens(currentShift)
+          currentShift += 1
+        }
+      }
+    }
+
+    texInfo
+  }
+
+  def parseMTL(mtlFile: Array[String]): scala.collection.Map[String, Material] = {
+    val mats: Map[String, Material] = Map()
+    var curMat: Option[Material] = None
+
+    def mat(): Material = curMat match {
+      case Some(cur) => cur
+      case None      => throw new GLException("No material currently selected")
+    }
+
+    def flushCurMat(): Unit = curMat match {
+      case Some(cur) => {
+        mats += (cur.name -> cur)
+        curMat = None
+      }
+      case None =>
+    }
+
+    for (currentLine <- mtlFile) {
+      val index = currentLine.indexOf("#")
+      val line = if (index < 0) currentLine else currentLine.substring(0, index).trim()
+
+      val tokens = line.split(" ", -1)
+
+      (tokens(0).toLowerCase(), if (tokens.size >= 2) Some(tokens(1).toLowerCase()) else None) match {
+        case ("newmtl", _) if (tokens.size >= 2) => {
```

Comment:
All the coding styles I've seen agree on *not* putting `{` on a multi-statement `case`.